framework/base/Model.php

```diff
@@ -773,7 +773,7 @@ public function activeAttributes()
         if (!isset($scenarios[$scenario])) {
             return [];
         }
-        $attributes = $scenarios[$scenario];
+        $attributes = array_unique($scenarios[$scenario]);
         foreach ($attributes as $i => $attribute) {
```

Comment:
```php
        return array_map(function($attribute) {
            return ltrim($attribute, '!');
        }, array_unique($scenarios[$scenario]));
```

or make new static method `Model::attributeNames(array $attributes)` and invoke it here and on validator